com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/v2/ApplicationMetadataBuilder.java

```diff
@@ -0,0 +1,88 @@
+package com.sap.cloud.lm.sl.cf.core.cf.v2;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+import org.cloudfoundry.client.v3.Metadata;
+
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.MtaMetadataAnnotations;
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.MtaMetadataLabels;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaModule;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaResource;
+import com.sap.cloud.lm.sl.cf.core.model.ImmutableDeployedMtaModule;
+import com.sap.cloud.lm.sl.cf.core.model.ImmutableDeployedMtaResource;
+import com.sap.cloud.lm.sl.cf.core.model.SupportedParameters;
+import com.sap.cloud.lm.sl.cf.core.util.NameUtil;
+import com.sap.cloud.lm.sl.common.util.JsonUtil;
+import com.sap.cloud.lm.sl.mta.model.DeploymentDescriptor;
+import com.sap.cloud.lm.sl.mta.model.Module;
+import com.sap.cloud.lm.sl.mta.model.ProvidedDependency;
+import com.sap.cloud.lm.sl.mta.model.Resource;
+
+public class ApplicationMetadataBuilder {
+
+    public static Metadata build(DeploymentDescriptor deploymentDescriptor, Module module, List<ResourceAndResourceType> moduleResources,
```

Comment:
Please, keep the naming - if you are using resource user resource as a whole if service then service, do not mix them. In this case, the resource should be used.